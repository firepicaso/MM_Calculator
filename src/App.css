body {
  background-color: hsl(0, 0%, 93%);
  min-height: 100vh;
}

header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 2rem;
}

.nav-list {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.1rem;
  font-weight: 600;
  display: flex;
  list-style: none;
  gap: 0.5rem;
}

.nav-text {
  color: rgb(20, 97, 197);
}

.homepage {
  margin: 2rem;
}

.app-container {
  display: flex;
  justify-content: space-between;
  margin: 2rem;
}

.quote-container {
  display: flex;
  justify-content: center;
}

.quote {
  margin: 2rem;
}

h1 {
  font-family: Arial, Helvetica, sans-serif;
  color: rgb(58, 58, 58);
}

h2 {
  font-family: Arial, Helvetica, sans-serif;
  color: rgb(58, 58, 58);
}

h4 {
  font-family: Arial, Helvetica, sans-serif;
  color: rgb(172, 133, 8);
}

.calculator {
  display: grid;
  max-width: 300px;
  margin-right: 2rem;
}

.button-row {
  display: flex;
  justify-content: space-between;
}

.button-row :last-child {
  background-color: rgb(253, 155, 36);
}

button {
  width: 100%;
  border: 1px solid rgb(195, 195, 195);
  font-size: 18px;
  padding: 1rem;
  cursor: pointer;
}

.display {
  display: flex;
  align-items: center;
  justify-content: end;
  background-color: #8f8f8f;
  font-size: 22px;
  padding: 0.5rem 0.5rem;
  color: white;
}

.narrow {
  width: 140px;
}
