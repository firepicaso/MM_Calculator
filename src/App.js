import React from 'react';
import {
  BrowserRouter as Router, Route, Routes, Link,
} from 'react-router-dom';
import Calculator from './components/Calculator';
import Home from './components/home';
import Quote from './components/Quote';
import './App.css';

export default function App() {
  return (
    <Router>
      <div className="App">
        <header>
          <h1>Math Magicians</h1>
          <nav>
            <ul>
              <li>
                <Link to="/">Home</Link>
              </li>
              <li>
                <Link to="/calculator">Calculator</Link>
              </li>
              <li>
                <Link to="/quote">Quote</Link>
              </li>
            </ul>
          </nav>
        </header>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/calculator" element={<Calculator />} />
          <Route path="/quote" element={<Quote />} />
        </Routes>
      </div>
    </Router>
  );
}
