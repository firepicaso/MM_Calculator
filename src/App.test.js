import { render, screen } from '@testing-library/react';
import App from './App';
import '@testing-library/jest-dom/extend-expect';

test('testing Apppage component', () => {
  const { container } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();
});

test('simulating user interaction', () => {
  render(<App />);

  const quoteElement = screen.getByText('Math Magicians');
  expect(quoteElement).toBeInTheDocument();
});
