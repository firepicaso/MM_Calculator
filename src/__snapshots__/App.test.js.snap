// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Apppage component 1`] = `
<div
  class="App"
>
  <header
    class="header"
  >
    <h1>
      Math Magicians
    </h1>
    <nav>
      <ul
        class="nav-list"
      >
        <li>
          <a
            class="nav-text"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          |
        </li>
        <li>
          <a
            class="nav-text"
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li>
          |
        </li>
        <li>
          <a
            class="nav-text"
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    class="homepage"
  >
    <h2>
      Welcome to our page!
    </h2>
    <div
      class="welcome"
    >
      <p>
        Welcome to our innovative calculator app that is about sparking your curiosity
      </p>
      <p>
        Our calculator is your reliable companion on your mathematical journey.
      </p>
      <p>
        You will find that math is an adventure waiting to be explored.
      </p>
    </div>
  </div>
</div>
`;
